api/storage.go

```diff
@@ -0,0 +1,10 @@
+package api
+
+type Storage struct {
+	Name          string
+	Driver        string
+	DriverOptions map[string]string
+	Source        string
+	Fstype        string
+	MoutOptions   []string
+}
```

Comment:
So `Storage` represents e.g. an RBD or ceph volume attached to a VM/Pod? It's a pod resource that can be then mounted in one or more containers? Is that a correct description?
And so `sandbox.AddStorage()` is the part that attaches the storage to the VM/Pod, without mounting it in any container?

Reply:
Yes, your description is correct. Just to be clear, `Storage` is not limited to RBD. `sandbox.AddStorage()` is supposed to call the hypervisor hotplug code. It won't mount the storage in any container because `sandbox.AddStorage()` can be called before `sandbox.AddContainer` thus there might be no containers in the sandbox at all.

Reply:
So do you see any reason why it could not be handled as part of the sandbox/pod creation step? Or is `addStorage` potentially being called outside of the pod/sandbox creation step?

Reply:
I'm not sure if I understand your question. In CRI, sandbox creation does not include all container specs. `sandbox.AddStorage` is supposed to be called when handling the CRI `CreateContainer()` request.

Reply:
Right, so CRI `CreateContainer()` will eventually call into virtcontainers' `CreateContainer()` or `pod.CreateContainer()`: Can we not export `sandbox.AddStorage()` and make it an internal call of virtcontainers `CreateContainer()` or `pod.CreateContainer()`?

Reply:
As I mentioned [above](https://github.com/kata-containers/runtime/pull/32#discussion_r173491512) to reply to @sboeuf, exporting these hotplug APIs will allow upper layer to prepare rootfs/volumes/nics in parallel that can save us hundreds of milliseconds when creating a new container.

Another reason to not reject `AddStorage` is that we will need to export `AddNic` anyway to support CNI even in the kata runtime CLI case, assuming @WeiZhang555 still wants the `kata-cli interface` subcommand similar to `runv interface` he implemented in runv. IOW, the hotplug APIs are not only useful for CRI native support. They are useful for kata-cli as well.